src/Admin/BlockAdmin.php

```diff
@@ -233,4 +202,69 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
             return $options['template'];
         }
     }
+
+    private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
+    {
+        $service = $this->blockManager->get($block);
+        $blockType = $block->getType();
+
+        if (!$service instanceof BlockServiceInterface) {
+            throw new \RuntimeException(sprintf(
+                'The block "%s" is not a valid %s',
+                $block->getType(),
+                BlockServiceInterface::class
+            ));
+        }
+
+        if ($service instanceof EditableBlockService) {
+            $blockMapper = new PageFormMapper($formMapper);
+            if ($block->getId() > 0) {
+                $service->configureEditForm($blockMapper, $block);
+            } else {
+                $service->configureCreateForm($blockMapper, $block);
+            }
+        } else {
+            @trigger_error(
+                sprintf(
+                    'Editing a non %s block service is deprecated since sonata-project/page-bundle 3.x and will not be allowed with version 4.0.',
```

Comment:
```suggestion
                    'Editing a block service which doesn\'t implement %s is deprecated since sonata-project/page-bundle 3.x and will not be allowed with version 4.0.',
```